### DISCUSSION TITLE
Is the GPL2 public license model in this project problematic when using NsDepCop in proprietary projects?

### DISCUSSION THREAD

**Discussion by HHobeck** [realvizu/NsDepCop#72] [Q&A] [answered]
Hi there.

I'm wondering how the GPL2 public license of this project is affecting the license model of other projects. Let’s say I'm using it for a proprietary project to ensure my namespace references on compile time. Do I need to publish the proprietary project under the same license model? If not are any other obligations existing?

Thank you for answering.

BTW: Very nice tool. Thank you!

**Comment by realvizu** [comment, marked as answer]
Hi!

GPL2 allows you to use NsDepCop for commercial purposes, so you can use it for developing/analyzing any other (propriatery) projects.
You don't have to publish your propriatery project under GPL2 because of using NsDepCop at development time.

If you were to create and publish derived products from NsDepCop itself, that's when the obligation to license it under GPL2 would come into play.

Hope this helps.
And I'm glad you like NsDepCop :)
